hashednode.go

```diff
@@ -78,11 +101,15 @@ func (*HashedNode) Serialize() ([]byte, error) {
 }
 
 func (n *HashedNode) Copy() VerkleNode {
-	if n.commitment == nil {
-		panic("nil commitment")
+	if n.serialized == nil {
+		panic("invalid hashed node")
 	}
-	c := &HashedNode{commitment: make([]byte, len(n.commitment))}
-	copy(c.commitment, n.commitment)
+
+	c := &HashedNode{
+		serialized: make([]byte, len(n.serialized)),
+	}
+	copy(c.serialized, n.serialized)
+
 	return c
 }
```

Comment:
That's it... not that wild change, but very convoluted to figure out via benchmarks from go-ethereum that this path was very hot.